tests/Builder/BuilderEncoderTest.php

```diff
@@ -279,6 +279,60 @@ public function testRedactStage(): void
         $this->assertSamePipeline($expected, $pipeline);
     }
 
+    public function testUnwind(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::unwind('$foo'),
+            Stage::unwind('$foo.bar'),
+            Stage::unwind(Expression::arrayFieldPath('foo.bar.baz')),
+        );
+
+        $expected = [
+            ['$unwind' => '$foo'],
+            ['$unwind' => '$foo.bar'],
+            ['$unwind' => '$foo.bar.baz'],
+        ];
+
+        $this->assertSamePipeline($expected, $pipeline);
+    }
+
+    public function testComplexGroupStage(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::group(
+                _id: null,
+                results: Accumulator::push(
+                    Expression::arrayToObject([[
+                       'k' => Expression::toString(Expression::fieldPath('key')),
+                       'v' => Expression::fieldPath('value'),
+                    ]]),
```

Comment:
It's strictly equivalent. But in order to match the expected pipeline, it have to be an object. Maybe `assertSamePipeline` need to be improved to accept array or object equally.
```suggestion
                    Expression::arrayToObject([
                        object(
                            k: Expression::toString(Expression::fieldPath('key')),
                            v: Expression::fieldPath('value'),
                        ),
                    ]),
```

Reply:
Yes it should not be necessary to change the pipeline itself to make the assert method happy 👌🏻

Reply:
This is solved by using a BSON round-trip. See https://github.com/mongodb/mongo-php-builder/pull/13